File: The Core/knapsackLight.cs
int knapsackLight(int value1, int weight1, int value2, int weight2, int maxW) {
    
    if(maxW >= weight1 + weight2){
        return value1 + value2;
    }
    
    if(value1 >= value2){
        if(weight1 <= maxW){
            return value1;
        }
        else if(weight2 <= maxW){
            return value2;
        }
    }  
    
    if(value2 >= value1){
        if(weight2 <= maxW){
            return value2;
        } else if (weight1 <= maxW){
            return value1;
        }
    }
    
    return 0;   
}

File: The Core/At The Crossroads/10. Knapsack Light/knapsackLight.lua
function knapsackLight(value1, weight1, value2, weight2, maxW)
    
    if maxW >= weight1 + weight2 then
        return value1 + value2
    end
    
    if value1 >= value2 then
        if weight1 <= maxW then
            return value1
        elseif weight2 <= maxW then
            return value2
        end
    end
    
    if value2 >= value1 then
        if weight2 <= maxW then
            return value2
        elseif weight1 <= maxW then
            return value1
        end
    end
    
    return 0           
end